### ISSUE TITLE
Yet another valid wasm module that fails validation

### ISSUE THREAD

**Issue by fitzgen** [bytecodealliance/wasm-tools#85]
[test.wasm.gz](https://github.com/bytecodealliance/wasm-tools/files/5104531/test.wasm.gz)

```wasm
(module
  (type (;0;) (func (result i32)))
  (type (;1;) (func (param i32 i32)))
  (import "" "" (table (;0;) 4278190080 funcref))
  (func (;0;) (type 1) (param i32 i32)
    call 1
    return
    local.get 1
    loop (param i32 i32)  ;; label = @1
      loop (param i32 i32)  ;; label = @2
        call 0
      end
    end
    unreachable)
  (func (;1;) (type 0) (result i32)
    unreachable))
```

fails with:

```
type mismatch: stack size does not match block type (at offset 119)
```

Once again, in unreachable code (following the unconditional `return`).

cc @yurydelendik

**Comment by fitzgen** [comment]
> I think this one is addressed by #84

I can repro even after #84 was merged

**Comment by fitzgen** [comment]
I wouldn't be surprised if there was another code path that needed to check loop vs other kinds of blocks to get params or results,
